components/portal/widgets/controllers.js

```diff
@@ -403,152 +403,133 @@ define(['angular'], function(angular) {
-        $scope.templateStatus = 'upcoming';
-      // If filter returns -1, the period has ended
-      } else if ($filter('filterDifferenceFromDate')(
-        $scope.activePeriodEndDate) === -1) {
-        $scope.templateStatus = 'ended';
+      var now = new Date();
+      var templateIndex = indexWithinTemplateRange(now, callsToAction);
+      if (-1 < templateIndex) {
+        displayTimeSensitiveContent(callsToAction[templateIndex]);
       }
+    };
 
-      // If we somehow ended up with an unset template status,
-      // don't show time sensitive content
-      if ($scope.templateStatus === '') {
-        $log.log('Date range didn\'t meet any of the criteria for '
-          + 'showing time-sensitive content');
-      } else {
-        $scope.showTimeSensitiveContent = true;
+    var resolveDate = function(str, defaultEndOfDay) {
+      var result = null;
+      if (str) {
+        var dateStr =
+          $filter('filterForDateWithYear')(str);
+        if (dateStr) {
+          if (dateStr.indexOf('T') < 0) {
+            if (defaultEndOfDay) {
+              dateStr += 'T23:59:59';
+            } else {
+              dateStr += 'T00:00:00';
+            }
+          }
+          result = new Date(dateStr);
+        }
       }
+      return result;
     };
 
-    var evaluateCallsToAction = function(callsToAction) {
+    var indexWithinTemplateRange = function(now, callsToAction) {
+      var result = -1;
       // Check if today falls within any of the provided date ranges
-      angular.forEach(callsToAction, function(value, index) {
-        var templateSwitchOnDate = '';
-        var templateSwitchOffDate = '';
-        var hasPaddedDates = {
-          start: false,
-          end: false,
-        };
-        // Check if active date range provided
-        if (value.activeDateRange) {
-          // Check for a start date
-          if (value.activeDateRange.templateLiveDate) {
-            // Use go live date if provided
-            templateSwitchOnDate = $filter('filterForDateWithYear')(
-              value.activeDateRange.templateLiveDate);
-            hasPaddedDates.start = true;
-          } else if (value.activeDateRange.takeActionStartDate) {
-            // Fall back on takeActionStartDate if needed
-            templateSwitchOnDate = $filter('filterForDateWithYear')(
-              value.activeDateRange.takeActionStartDate);
-          } else {
-            $log.log($scope.widget.fname +
-              ' said it has variable content, but was missing a '
-              + ' start date.');
-            showBasicWidget();
+      angular.forEach(callsToAction, function(callToAction, index) {
+        if (result === -1) { // We haven't found a call to action
+          var startDate =
+            resolveDate(callToAction.activeDateRange.templateLiveDate);
+          var endDate =
+            resolveDate(callToAction.activeDateRange.templateRetireDate, true);
+          if (startDate && startDate < now // Template started
```

Comment:
again moment is amazing: [`moment.isBefore`](https://momentjs.com/docs/#/query/is-between/)